Assets/Scripts/MineManager.cs

```diff
@@ -0,0 +1,267 @@
+            } while (scoreGrid[x, y] == -1);
+
+            scoreGrid[x, y] = -1;
+        }
+        
+        // define the state and GameObject of grids
+        for (int i = 0; i < gridSize; i++)
+        {
+            for (int j = 0; j < gridSize; j++)
+            {
+                gridStates[i, j] = State.Init;
+                GameObject newBox = Instantiate(boxs[(i+j)%2]);
+                newBox.transform.position = new Vector3((i-gridSize/2f+0.5f) *10f /(gridSize+1), (j-gridSize/2f+0.5f) *10f /(gridSize+1), 0);
+                newBox.transform.localScale = new Vector3(10f /(gridSize+1), 10f /(gridSize+1), 0);
+                newBox.name = "box" + i + "_" + j;
+                newBox.GetComponent<MineDisplay>().boxLocation = new Vector2Int(i, j);
+            }
+        }
+
+        for (int i = 0; i < gridSize; i++)
+        {
+            for (int j = 0; j < gridSize; j++)
+            {
+                if (scoreGrid[i, j] != -1)
+                {
+                    scoreGrid[i, j] = CheckNeighbor(new Vector2Int(i, j));
+                }
+            }
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (gameState != GameState.Playing)
+        {
+            time += Time.deltaTime;
+            if (time > delayTime)
+            {
+                gameState = GameState.Playing;
+                SceneManager.LoadScene("Scenes/End");
+            }
+        }
+    }
+    
+    int CheckNeighbor(Vector2Int loc)
+    {
+        int counter = 0;
+        for (int i = -1; i < 2; i++)
+        {
+            if (loc.x + i < 0 || loc.x + i >= gridSize)
+            {
+                continue;
+            }
+            for (int j = -1; j < 2; j++)
+            {
+                if (loc.y + j >= 0 && loc.y + j < gridSize)
+                {
+                    if (scoreGrid[loc.x + i, loc.y + j] ==-1)
+                    {
+                        counter += 1;
+                    }
+                }
+            }
+        }
+        return counter;
+    }
+
+    public void ChangeState(Vector2Int loc, State state)
+    {
+        gridStates[loc.x, loc.y] = state;
+        Changelist.Add((loc,state));
+        if (state == State.Opened)
+        {
+            opened += 1;
+            Debug.Log(loc+": "+opened);
+            if (scoreGrid[loc.x, loc.y] == -1)
+            {
+                Debug.Log("end game lose");
+                gameState = GameState.Lose;
+            }
+            if (opened == gridSize * gridSize - mineCount)
+            {
+                Debug.Log("end game win");
+                gameState = GameState.Win;
+            }
+        }
+    }
+
+    public void UpdateGridDisplay()
+    {
+        
+        while (Changelist.Count > 0)
+        {
+            Vector2Int loc= Changelist[0].Item1;
+            State s = Changelist[0].Item2;
+            GameObject box = GameObject.Find("box" + (loc.x) + "_" + (loc.y));
```

Comment:
this should be stored in an array.
NEVER LOOK THINGS UP VIA STRING IF YOU CAN AVOID IT